src/Nethermind/Nethermind.Serialization.Rlp/RlpStream.cs

```diff
@@ -358,7 +358,7 @@ public void Encode(long value)
 
         public void Encode(ulong value)
         {
-            Encode(value, 8);
+            Encode((UInt256)value, 8);
```

Comment:
why do we need this cast?

Reply:
ambiguous between UInt256 and BigInteger, BigInteger needed for negative values when encoding long